server/routers/viewer.tsx

```diff
@@ -0,0 +1,81 @@
+import { TRPCError } from "@trpc/server";
+
+import { createRouter } from "../createRouter";
+
+// routes only available to authenticated users
+export const viewerRouter = createRouter()
+  // check that user is authenticated
+  .middleware(({ ctx, next }) => {
+    const { user } = ctx;
+    if (!user) {
+      throw new TRPCError({ code: "UNAUTHORIZED" });
+    }
+
+    return next({
+      ctx: {
+        ...ctx,
+        // session value is known to be non-null now
+        user,
+      },
+    });
+  })
+  .query("me", {
```

Comment:
How should we go about splitting these into files?

Reply:
The same way we do on the app/root router to separate `viewer` in it's own file

https://github.com/calendso/calendso/blob/fb10bea04a0747e8abe005591b2780939c2da676/server/routers/_app.ts#L13-L24

**One can either do** 

```ts
export const myRouter = createRouter()
  .merge("sub.", subRouter); // <-- prefixes all procedures with `sub.`
```
**Or**

```ts
export const myRouter = createRouter()
  .merge(subRouter); // <-- does not prefix any procedures
```

And this works on the `viewerRouter` too, so if that file got insanely big or if we wanted to do distinct parts separately, they could be handled in their own file